qa/rpc-tests/mempool_size_limit.py

```diff
@@ -0,0 +1,593 @@
+
+        print("Starting actual test")
+        # Send chained txs first
+        tx_sent = 0
+        mark_logs("Sending chained transactions", self.nodes, DEBUG_MODE)
+        ctransactions = []
+        while (len(ctxs) > 0):
+            tx = ctxs.pop(0)['hex']
+            ctransactions.append(self.nodes[0].sendrawtransaction(tx, True))
+            txid = self.nodes[0].decoderawtransaction(tx)['txid']
+            txinfo = self.nodes[0].getrawmempool(True)[txid]
+            feerate = txinfo['fee'] / txinfo['size']
+            tx_sent += len(tx)//2
+
+        mark_logs("Filling mempools with more transactions...", self.nodes, DEBUG_MODE)
+        tx_sent_size = 0
+        feerates = {}
+        minfeerate = MAX_FEE
+        mininput = MAX_FEE
+        last_tx_size = 0
+        usage = int(self.nodes[0].getmempoolinfo()['bytes'])
+        while (usage < NODE0_LIMIT_B - (last_tx_size + 10000)):
+            assert(len(txs) > 0)
+            tx = txs.pop(len(txs)//2) # pick from the middle, use avg fee
+            tx_hex = tx['hex']
+            tx_input = tx['fee']
+            txid = self.nodes[0].decoderawtransaction(tx_hex)['txid']
+            self.nodes[0].sendrawtransaction(tx_hex, True)
+            txinfo = self.nodes[0].getrawmempool(True)[txid]
+            feerate = txinfo['fee'] / txinfo['size']
+            feerates[txid] = feerate
+            if (feerate < minfeerate):
+                minfeerate = feerate
+            if (tx_input < mininput):
+                mininput = tx_input
+            usage = int(self.nodes[0].getmempoolinfo()['bytes'])
+            last_tx_size = len(tx_hex)//2
+            assert_equal(last_tx_size, txinfo['size'])
+            #assert_equal(Decimal("7.5") - tx_input, txinfo['fee'])
+            tx_sent += last_tx_size
+
+        print(f"Mempool almost full: {usage} out of {NODE0_LIMIT_B}")
+        self.assert_limits_enforced()
+
+        mark_logs("Sending low fee transction, expecting failure", self.nodes, DEBUG_MODE)
+        try:
+            tx = txs.pop(0)
+            tx_hex = tx['hex']
+            tx_input = tx['fee']
+            assert(tx_input <= mininput)
+            txid = self.nodes[0].decoderawtransaction(tx_hex)['txid']
+            self.nodes[0].sendrawtransaction(tx_hex, True)
+            assert(False)
+        except JSONRPCException as e:
+            print("Ok")
+
+        self.assert_limits_enforced()
+        mpool = self.nodes[0].getrawmempool()
+        assert(txid not in mpool)
+
+        mark_logs("Sending high fee transaction, expecting success", self.nodes, DEBUG_MODE)
+        try:
+            tx = txs.pop()
+            tx_hex = tx['hex']
+            tx_input = tx['fee']
+            assert(tx_input > mininput)
+            txid = self.nodes[0].decoderawtransaction(tx_hex)['txid']
+            self.nodes[0].sendrawtransaction(tx_hex, True)
+            print("Ok")
+        except JSONRPCException as e:
+            assert(False)
+
+        prev_mpool = mpool
```

Comment:
What about checking the evicted tx is actually the one expected?
After keeping track of tx with minfeerate in previous filling loop, with something like
```
if (feerate < minfeerate):
  minfeerate = feerate
  minfeeratetxid = txid
```
you could then check
```suggestion
prev_mpool = mpool
assert(minfeeratetxid in mpool)
mpool = self.nodes[0].getrawmempool()
assert(minfeeratetxid not in mpool)
```

Reply:
Or possibly just extracting minimum before the assertions:
`min(feerates, key=feerates.get)`

Reply:
Agreed to handle this in a separate PR.